quinn-proto/src/address_discovery.rs

```diff
@@ -0,0 +1,128 @@
+//! Address discovery types from
+//! <https://datatracker.ietf.org/doc/draft-seemann-quic-address-discovery/>
+
+use crate::VarInt;
+
+pub(crate) const TRANSPORT_PARAMETER_CODE: u64 = 0x9f81a176;
+
+/// The role of each participant.
+///
+/// When enabled, this is reported as a transport parameter.
+#[derive(PartialEq, Eq, Clone, Copy, Debug, Default)]
+pub(crate) enum Role {
```

Comment:
It seems like we go to a lot of trouble to convert this to and from a pair of `bool`s. Would it it be simpler to represent it that way in the first place?